src/Hl7.Fhir.Base/Specification/Terminology/MimeTypeTerminologyService.cs

```diff
@@ -41,8 +41,9 @@ public async T.Task<Parameters> ValueSetValidateCode(Parameters parameters, stri
             parameters.CheckForValidityOfValidateCodeParams();
 
             var validCodeParams = new ValidateCodeParameters(parameters);
+            var valueSetUri = validCodeParams?.Url?.Value != null ? new Canonical(validCodeParams?.Url?.Value)?.Uri : null;
 
-            if (validCodeParams?.Url?.Value is not null && validCodeParams?.Url?.Value != MIMETYPE_VALUESET)
+            if (valueSetUri is not null && valueSetUri != MIMETYPE_VALUESET_R4_AND_UP && valueSetUri != MIMETYPE_VALUESET_STU3)
```

Comment:
Actually the condition `valueSetUri is not null` is not really needed here: `null !=  "http://hl7.org/fhir/ValueSet/mimetypes"` will also do the trick